src/main/java/com/artipie/api/KeyStore.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.api;
+
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.blocking.BlockingStorage;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PfxOptions;
+import java.util.List;
+
+/**
+ * Key store.
+ * @since 0.26
+ */
+public abstract class KeyStore {
+    /**
+     * YAML node name for `enabled` yaml section.
+     */
+    private static final String NODE_ENABLED = "enabled";
+
+    /**
+     * YAML node name for `path` yaml section.
+     */
+    private static final String PATH = "path";
+
+    /**
+     * YAML node name for `password` yaml section.
+     */
+    private static final String PASSWORD = "password";
+
+    /**
+     * YAML node name for `alias` yaml section.
+     */
+    private static final String ALIAS = "alias";
+
+    /**
+     * YAML node name for `aliasPassword` yaml section.
+     */
+    private static final String ALIAS_PASSWORD = "alias-password";
+
+    /**
+     * YAML-configuration of key store.
+     */
+    private final YamlMapping yml;
+
+    /**
+     * Ctor.
+     * @param yaml YAML.
+     */
+    public KeyStore(final YamlMapping yaml) {
+        this.yml = yaml;
+    }
+
+    /**
+     * Checks if SSL is enabled.
+     * @return True is SSL enabled.
+     */
+    public boolean enabled() {
+        return Boolean.parseBoolean(this.yml.string(KeyStore.NODE_ENABLED));
+    }
+
+    /**
+     * Create KeyStore instance.
+     * @param yaml Settings of key store
+     * @return KeyStore
+     */
+    @SuppressWarnings("PMD.ProhibitPublicStaticMethods")
+    public static KeyStore create(final YamlMapping yaml) {
+        final List<KeyStore> keystores = List.of(
+            new JksKeyStore(yaml), new PemKeyStore(yaml), new PfxKeyStore(yaml)
+        );
+        for (final KeyStore keystore : keystores) {
+            if (keystore.isConfigured()) {
+                return keystore;
+            }
+        }
```

Comment:
hm, so if there are several key store types configured, we simply use any of them... I'm not sure, that's the best course of actions, as it's not obvious to user how we choose what type to use...

Reply:
@olenagerasimova if user specifies several will be used only one, we need to document usage SSL with examples and explanation.
I think that we met pitfail of simplicity of YAML because YAML does not support validation if we compare it with XML/XSD-schemas where XSDchema allows to restrict by only one of alternatives. May YAML-has some validation it would be good for us to validate artipie-configurations before storing them.

Reply:
Documentation sure should be added in the next step, when this [PR](https://github.com/artipie/artipie/pull/1158) is merged.